src/modules/main/PlayerView.tsx

```diff
@@ -1,23 +1,159 @@
+  // labels will be null only before setting the state as we cannot render all labels within container if not settled yet
+  const [labels, setLabels] = useState<null | Label[]>(null);
+
+  const { data: image } = hooks.useAppSettings({
+    name: SettingsKeys.File,
+  });
+  const [isSubmitted, setIsSubmitted] = useState(false);
+
+  const answersAppData = appData?.filter(
+    ({ type }) => type === ANSWER_SUBMISSION_TYPE,
+  );
+  const settingLabels = appSettings?.[0].data.labels;
+  const lastVersion = answersAppData?.[answersAppData.length - 1];
+  const answers = lastVersion?.data.answers;
+
+  const retry = (): void => {
+    if (settingLabels) {
+      const answered = settingLabels?.map((label) => ({
+        expected: label,
+        actual: null,
+      }));
+
+      setAnsweredLabels(answered);
+      setLabels(settingLabels);
+    }
+  };
+
+  const save = (): void => {
+    const submittedAnswers = answeredLabels.map(({ expected, actual }) => ({
+      expectedId: expected.id,
+      actualId: actual?.id,
+    }));
+    saveAppData({
+      data: { answers: submittedAnswers },
+      type: ANSWER_SUBMISSION_TYPE,
+    });
+  };
+
+  useEffect(() => {
+    if (settingLabels) {
+      if (answers) {
+        const answered = answers?.map(
+          ({ expectedId, actualId }: SubmittedAnswer) => ({
+            expected: settingLabels.find(
+              ({ id }) => id === expectedId,
+            ) as Label,
+            actual: settingLabels.find(({ id }) => id === actualId) || null,
+          }),
+        );
+
+        setAnsweredLabels(answered);
+        setLabels(
+          settingLabels.filter(
+            ({ id }) =>
+              !answers.find(({ actualId }: SubmittedAnswer) => actualId === id),
+          ),
+        );
+      } else {
+        const answered = settingLabels.map((label) => ({
+          expected: label,
+          actual: null,
+        }));
+
+        setAnsweredLabels(answered);
+        setLabels(settingLabels);
+      }
+    }
+  }, [answers, settingLabels]);
+
+  useEffect(() => {
+    const answered = answers?.map(
+      ({ expectedId, actualId }: SubmittedAnswer) => ({
+        expected: settingLabels?.find(({ id }) => id === expectedId) as Label,
+        actual: settingLabels?.find(({ id }) => id === actualId) || null,
+      }),
+    );
+    setIsSubmitted(isEqual(answeredLabels, answered));
+  }, [answeredLabels, answers, settingLabels]);
+
+  if (isLoading) {
+    return <CircularProgress />;
+  }
+
+  if (!image || !settingLabels) {
+    return <Alert severity="error">{t(APP.UNCONFIGURED_ITEM)}</Alert>;
+  }
 
   return (
-    <div data-cy={PLAYER_VIEW_CY}>
-      Player as {permission}
-      <Box p={2}>
-        <Typography>Members</Typography>
-        <pre>{JSON.stringify(members, null, 2)}</pre>
-      </Box>
-    </div>
+    <Box data-cy={PLAYER_VIEW_CY}>
+      <Container>
+        <Stack spacing={2} padding={2}>
+          <Box className="KKK">
```

Comment:
What is the className `KKK`? Maybe extract it in a well named const, or at least add a comment :).